src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -148,31 +148,32 @@ export class TerminalPanel extends Panel {
 				// Drop selection and focus terminal on Linux to enable middle button paste when click
 				// occurs on the selection itself.
 				this._terminalService.getActiveInstance().focus();
-			} else if (event.which === 3) {
-				if (this._terminalService.configHelper.getRightClickCopyPaste()) {
-					let terminal = this._terminalService.getActiveInstance();
-					if (terminal.hasSelection()) {
-						terminal.copySelection();
-						terminal.clearSelection();
-					} else {
-						terminal.paste();
-					}
+			}
+		}));
+		this._register(DOM.addDisposableListener(this._parentDomElement, 'contextmenu', (event: MouseEvent) => {
+			if (this._terminalService.configHelper.getRightClickCopyPaste()) {
```

Comment:
I was thinking I can use the if condition of `getRightClickCopyPaste` in the `mousedown` event and revert the if condition of `getRightClickCopyPaste` in the `contextmenu` event. Not sure how I can `cancel` the triggering of the event any other way, maybe I'm missing something obvious here.